kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -218,7 +246,9 @@ public void onClosed(WebSocket webSocket, int code, String reason) {
         webSocketClosed.set(true);
         LOGGER.debug("Exec Web Socket: On Close with code:[{}], due to: [{}]", code, reason);
         try {
-            closeExecutor();
+            if (cleanupOnClose.get()) {
```

Comment:
This will be false if the websocket is closed before the ExecWatch, and so `cleanUp()` won't be called. I think that `out` and `err` should be closed if applicable on WebSocket close, regardless of whether `EWSL.close()` has been called.

This would signal to `InputStreamPumper` that no more output is coming, and so that thread can complete.